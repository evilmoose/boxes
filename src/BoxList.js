import React, {useState} from "react";
import Box from "./Box";

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);
    
    const addBox = boxObj => {
        setBoxes(boxes => [...boxes, boxObj]);
    };
    
    const removeBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    };

    const boxComponent = boxes.map(box => (
        <Box
            key={box.id}
            id={box.id}
            width={box.width}
            height={box.height}
            handleRemove={removeBox}
            backgroundColor={box.backgroundColor}
        />
    ));

    return (
        <div>

        </div>
    );
}

export default BoxList;