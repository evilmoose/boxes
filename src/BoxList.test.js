import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import BoxList from './BoxList'; // Adjust the import path according to your file structure

describe('BoxList', () => {
  it('renders without crashing', () => {
    render(<BoxList />);
  });

  it('renders correctly and matches snapshot', () => {
    const tree = renderer.create(<BoxList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
