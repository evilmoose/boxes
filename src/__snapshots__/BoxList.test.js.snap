// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxList renders correctly and matches snapshot 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="backgroundColor"
        >
          Background Color
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <button
        id="newBoxButton"
      >
        Add a new box!
      </button>
    </form>
  </div>
</div>
`;
